src/fhirtypes/ElementDefinition.ts

```diff
@@ -1889,6 +1892,69 @@ export class ElementDefinition {
     return [];
   }
 
+  /**
+   * Returns an array of an ElementDefinition's unfolded children.
+   * @param {ElementDefinition} targetElement - The ElementDefinition being unfolded
+   * @returns {ElementDefinition[]} An array of the targetElement's children, with the IDs altered and
+   * the original property re-captured.
+   */
+
+  private cloneChildren(targetElement: ElementDefinition): ElementDefinition[] {
+    return targetElement?.children().map(e => {
+      const eClone = e.clone();
+      eClone.id = eClone.id.replace(targetElement.id, this.id);
+      eClone.structDef = this.structDef;
+      eClone.captureOriginal();
+      return eClone;
+    });
+  }
+
+  /**
+   * Checks a StructureDefinition's differential to determine if the profile element extension has been used on
+   * an element.
+   * @param {Object} profileJson - The json representation of this ElementDefinition's structDef
+   * @returns {boolean} True if the profile element extension is found on this elements profile property, false
+   * if the extension is not found
+   */
+  private hasProfileElementExtension(profileJson?: any): boolean | string {
```

Comment:
The return type here should just be `boolean` now. And also is `profileJson` still an optional argument? I think at this point, won't it always be passed?

I am also looking through this function, and I know before it was also handling unfolding on non-`BackboneElement` data types. Are there pieces of this function that could be simplified now that we know this is only getting called on `contentReference` elements? Or do you feel we can't/shouldn't make those simplifications.

Reply:
There definitely are simplifications that could be made to the function, but I thought that we would still want the function to identify the profile-element extension on non-`contentReference` elements if/when we go back and broaden support for the extension. That's why I kept `profileJson` as an optional parameter, since it's necessary for `contentReference` elements but not needed for regular elements.